src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockgrid/umbblockgridsortable.directive.js

```diff
@@ -0,0 +1,705 @@
+                } else if(dragY > currentContainerRect.bottom) {
+                    move(orderedContainerElements, -1);
+                }
+            }
+
+            function move(orderedContainerElements, newElIndex) {
+
+                newElIndex = newElIndex === -1 ? orderedContainerElements.length : newElIndex;
+                
+                const placeBeforeElement = orderedContainerElements[newElIndex];
+                if (placeBeforeElement) {
+                    // We do not need to move this, if the element to be placed before is it self.
+                    if(placeBeforeElement !== currentElement) {
+                        currentContainerElement.insertBefore(currentElement, placeBeforeElement);
+                    }
+                } else {
+                    currentContainerElement.appendChild(currentElement);
+                }
+
+                if(config.onChange) {
+                    config.onChange();
+                }
+            }
+
+
+            /** Removes an element from container and returns its items-data entry */
+            vm.getItemOfElement = function (element) {
+                if(!element) {
+                    return null;
+                }
+                return vm.items.find(entry => config.compareElementToModel(element, entry));
+            }
+            vm.removeItem = function (item) {
+                if(!item) {
+                    return null;
+                }
+                const oldIndex = vm.items.indexOf(item);
+                if(oldIndex !== -1) {
+                    return vm.items.splice(oldIndex, 1)[0];
+                }
+                return null;
+            }
+
+            vm.sync = function(element, fromVm) {
+
+                const movingItem = fromVm.getItemOfElement(element);
+                if(!movingItem) {
+                    console.error("Could not find item of sync item")
+                    return false;
+                }
+                if(vm.notifyRequestDrop({item: movingItem}) === false) {
+                    console.error("was not allowed for drop on this location.")
+                    return false;
+                }
+                if(fromVm.removeItem(movingItem) === null) {
+                    console.error("Sync could not remove item")
+                    return false;
+                }
+
+                /** Find next element, to then find the index of that element in items-data, to use as a safe reference to where the item will go in our items-data.
+                 * This enables the container to contain various other elements and as well having these elements change while sorting is occurring.
+                 */
+
+                // find next valid element (This assumes the next element in DOM is presented in items-data, aka. only moving one item between each sync)
+                let nextEl;
+                let loopEl = element;
+                while(loopEl = loopEl.nextElementSibling) {
```

Comment:
My local eslint is throwing an error on this because of the assignment in the while expression. Consider doing something like `while(loopEl)` and then inside the loop reassign `loopEl = loopEl.nextElementSibling`.

Reply:
I wrapped it in `( )` wouldn't that be okay? eslint is happy now :-D